core/blockchain.go

```diff
@@ -1191,13 +1194,16 @@ func (bc *BlockChain) insertChain(chain types.Blocks) (int, []interface{}, []*ty
 			return i, events, coalescedLogs, err
 		}
 		// Process block using the parent state as reference point.
+		t0 := time.Now()
 		receipts, logs, usedGas, err := bc.processor.Process(block, state, bc.vmConfig)
+		t1 := time.Now()
```

Comment:
Instead of measuring and updating, perhaps it would be cleaner to update on the fly?

```go
		start := time.Now()
		receipts, logs, usedGas, err := bc.processor.Process(block, state, bc.vmConfig)
		blockExecutionTimer.Update(time.Since(start))
```